src/interpreter/mod.rs

```diff
@@ -979,10 +994,23 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
         match value {
             Value::ByRef(ptr) => self.move_(ptr, dest, dest_ty),
             Value::ByVal(primval) => self.memory.write_primval(dest, primval),
+            Value::ByValPair(a, b) => {
+                self.memory.write_primval(dest, a)?;
+                let extra_dest = dest.offset(self.memory.pointer_size() as isize);
```

Comment:
This won't always be correct. We should inspect the `dest_ty` to figure out the second field offset. You can leave a `FIXME` for now.